app/code/community/Twispay/Tpay/Helper/Data.php

```diff
@@ -1,8 +1,1097 @@
+    } else {
+      return Mage::getStoreConfig('payment/tpay/stagingSiteId', Mage::app()->getStore()->getStoreId());
+    }
+  }
+
+
+  /**
+   * Function that extracts the value of the "apiKey" from
+   *  the config depending of the "liveMode" value.
+   */
+  public function getApiKey(){
+    if(1 == $this->getLiveMode()){
+      return Mage::getStoreConfig('payment/tpay/liveApiKey', Mage::app()->getStore()->getStoreId());
+    } else {
+      return Mage::getStoreConfig('payment/tpay/stagingApiKey', Mage::app()->getStore()->getStoreId());
+    }
+  }
+
+
+  /**
+   * Function that extracts the value of the "url"
+   *  depending of the "liveMode" value.
+   */
+  public function getUrl(){
+    if(1 == $this->getLiveMode()){
+      return $this->live_host_name;
+    } else {
+      return $this->stage_host_name;
+    }
+  }
+
+
+  /**
+   * Function that extracts the value of the "api url"
+   *  depending of the "liveMode" value.
+   */
+  public function getApiUrl(){
+    if(1 == $this->getLiveMode()){
+      return $this->live_api_host_name;
+    } else {
+      return $this->stage_api_host_name;
+    }
+  }
+  /************************** Config functions END **************************/
+
+
+
+  /************************** Api START **************************/
+  /**
+   * Function that extracts the status of an order from the
+   *  Twisplay platform.
+   * 
+   * @param orderId: The unique Twispay order ID of a recurring profile.
+   *
+   * @return string(server-status) In case of success.
+   *         bool(FALSE)           If case if failure.
+   */
+  public function getServerOrderStatus($orderId){
+    /* Get the config values. */
+    $apiKey = Mage::helper('tpay')->getApiKey();
+    Mage::Log(__FUNCTION__ . ': apiKey=' . print_r($apiKey, true), Zend_Log::DEBUG, $this->logFileName);
+    $url = Mage::helper('tpay')->getApiUrl();
+    Mage::Log(__FUNCTION__ . ': url=' . print_r($url, true), Zend_Log::DEBUG, $this->logFileName);
+
+    if ('' == $apiKey) {
+      Mage::Log(__FUNCTION__ . Mage::helper('tpay')->__(' Order details extraction failed: Incomplete or missing configuration.'), Zend_Log::ERR , $this->logFileName, /*forceLog*/TRUE);
+      return FALSE;
+    }
+
+    /* Create the URL. */
+    $url = $url . '/order/' . $orderId;
+
+    /* Make the server request. */
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "GET");
+    curl_setopt($ch, CURLOPT_HTTPHEADER, ['accept: application/json', 'Authorization: ' . $apiKey]);
+    /* Send the request. */
+    $response = curl_exec($ch);
+    curl_close($ch);
+    /* Decode the response. */
+    $response = json_decode($response);
+
+    /* Check if the response code is 200 and message is 'Success'. */
+    if ((200 == $response->code) && ('Success' == $response->message)) {

```

Comment:
We should also check if the response was decoded successfully with `json_decode`.

Reply:
Added the check.